### PR TITLE
add goreleaser workflow

### PR DESCRIPTION
**Pull request by shorsher**
Uses `goreleaser` to create new binary releases of `firefly-cli`. As a part of the release, a new homebrew tap will be created

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: goreleaser
+
+on:
+  push:
+    branches:
+      - 'main'
```
Comment (hfuss) [inline .github/workflows/release.yml:6]: Do we want it on push to main or only when a tag is made? At first I thought push to main made sense, but now I'm thinking we only want the latter bc goreleaser needs tag?

**Review by hfuss — APPROVED** [review]
LGTM from what I know of Actions and recall from the little bit I've used Goreleaser but I never messed w/ making brews.

File: .github/workflows/release.yml (hunk 1 above)
Comment (shorsher) [reply .github/workflows/release.yml:6]: Yeah I was thinking only when a tag is made is probably better, but I'll leave it up to @nguyer

**Comment by shorsher** [comment]
Still need to test the brew formula

**Pull request closed without merging**